### PR TITLE
[Fab 167] [Minor] Add functionality to support consul-stored IPsets

### PR DESCRIPTION
**Pull request by varadzhakov**
**Link to ticket**
https://bedegaming.atlassian.net/browse/FAB-167

**Purpose**
IP whitelisting is moving from hiera-defined to conan-defined configuration with the sole idea of making it easy

**Description of the proposed change**
 - Custom fact added to retrieve the ipsets stored in consul (configured through Conan in here http://conan.ad.bedegaming.com/#cm_settingbundles/82/edit)
 - Custom functions added to compile the puppet resource type definitions based on the hash of ipsets from consul

### CODE REVIEW

File: lib/facter/consul_ipsets.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        end
+
+    end
+
+    ipsetsGroupedByRuleAndPriority
+
+end
+
+ipsets = get_consul_ipsets("http://localhost:8500/v1/kv/config/bundle/ipsets")
```
Comment (robrankin) [inline lib/facter/consul_ipsets.rb]: hard coded to local agent?  might want to allow URL customization, could well be times when a node doesnt have a consul agent on it, and would want to GET the data from other node/server?

Comment (robrankin) [reply lib/facter/consul_ipsets.rb]: Actually same for specific K/V path... if you ever want to PR this back to the public module.

File: lib/puppet/parser/functions/get_ipsets_files.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        
+        ipsets = args[0]
+
+        ipset_files = {}
+
+        ipsets.each do |ipsetName, ips| 
+            
+            ipset_files["/opt/ipsets/#{ipsetName}.zone"] = {
+                "ensure"    => "file",
+                "owner"     => "root",
```
Comment (robrankin) [inline lib/puppet/parser/functions/get_ipsets_files.rb]: Owner and group should probably be manageable as well, some people may want to change that

Comment (varadzhakov) [reply lib/puppet/parser/functions/get_ipsets_files.rb]: Agree, will set that as a config in the hiera

**Pull request closed without merging**